src/pspm_con1.m

```diff
@@ -141,14 +141,14 @@ function pspm_con1(modelfile, connames, convec, datatype, deletecon, options)
           'data.stats are then used in a contrasts, producing an invalid results (NaN).']);
         data_stats_converted = data.stats;
         data_stats_converted(isnan(data_stats_converted)) = 0;
-        conval = conmat * data_stats_converted;
       else
         warning(['Calculated data.stats contain NaNs that are caused by unknown reasons. '...
           'However they were not used in the computation of the contrasts.']);
-        conval = conmat * data.stats;
+        data_stats_converted = data.stats;
       end
     end
   end
+  conval = conmat * data_stats_converted;
```

Comment:
No, this does not work, because data_stats_converted is rewritten at every iteration before the contrast value is actually written.
I suggest you focus on your deadline, and we arrange a meeting to come back to this issue together. In the meantime the release should be suspended.